RPIPs/RPIP-4.md

```diff
@@ -14,65 +14,68 @@ The Rocket Pool community at large agrees to use Snapshot voting for governance
 - an integer `power` which contains the ultimate voting power of that node
 
-`power` for `results` is determined in accordance with the following equation for voting: 
+`results.power` is determined in accordance with the following equation for voting: 
 
-`power = (int) 0.5 * √rpl * weight`, where `rpl` is the amount of RPL staked in that node, `weight` is the result of the weight calculation (see below), and the result is truncated into an integer.
+`results.power = (int) √rpl`, where `rpl` is the amount of RPL staked in that node, with the result truncated into an integer for simplicity.
 
-`weight` for is determined from the age of the node via:
+Any delegated vote `results.power` is summed with the voter's own `results.power` to achieve the final vote power for a single voter. Addresses which delegate their vote power are not included in the `results` map, but if a address that delegates its power instead votes directly, the address' `results.power` is no longer contributed towards the delegated address' power and is instead counted for the original address only.
 
-`weight = min( (currentDate - registrationDate) / 100, 1)` where `currentDate` is the date of the snapshot in UTC and `registrationDate` is the date on which the node was registered in UTC.
+Total vote power for each of the poll choices indicated in each `results.vote` entry is summed into an `rplTotalPower` array containing the total sum of power for each of the poll choices, with the index of each entry mapping to the integer corresponding to the vote choice. This is used to compute the outcome as described below.
 
-Any delegated vote `power` is summed with the voter's own `power` to achieve the final vote power for a single voter.
+#### Outcome
 
-Total vote power for `results` is summed into `rplTotalPower`.
+The `rplTotalPower` array is searched for the maximum value. The index of this maximum value corresponds to the winning proposal. If the winning choice corresponds to "abstain", however, then the proposal corresponding to the index of the second highest value is declared as the winner instead.
 
-#### Outcome
+#### Vetoing
+Given that the snapshot voting system is not comfortably secure (see Security Considerations below), there are two safeguard vetoes in place.
 
-All `power` values from the `rEthResults` map are summed into two integers based on each vote, `rEthYeaPower` and `rEthNayPower`, and all `power` values from the `rplResults` map are summed into two integers based on each vote, `rplYeaPower` and `rplNayPower`.
+The first veto consists of those with the power to bring forward a vote (see Implementation above) refusing to bring such a vote forward for consideration. Although potential changes can be technically implemented by the pDAO guardian regardless of vote, the pDAO guardian MUST NOT do so and any such action is considered illegitimate.
```

Comment:
```suggestion
The first veto consists of those with the power to bring forward a vote (see Implementation above) refusing to bring such a vote forward for consideration. Although potential changes can be technically implemented by the implementer (oDAO or  pDAO guardian depending on the proposal) regardless of vote, the implementer MUST NOT do so and any such action is considered illegitimate.

A second, more absolute, veto consists of the implementer refusing to implement a passed proposal.
```

Minor edit to note the "implementer role" varies based on proposal (spend is pDAO, contract stuff is oDAO, settings changes depend on what setting).